modules/structs/commit_status.go

```diff
@@ -16,13 +16,16 @@ const (
 	CommitStatusError CommitStatusState = "error"
 	// CommitStatusFailure is for when the CommitStatus is Failure
 	CommitStatusFailure CommitStatusState = "failure"
+	// CommitStatusWarning is for when the CommitStatus is Warning
+	CommitStatusWarning CommitStatusState = "warning"
 )
 
 var commitStatusPriorities = map[CommitStatusState]int{
 	CommitStatusError:   0,
 	CommitStatusFailure: 1,
 	CommitStatusPending: 2,
 	CommitStatusSuccess: 3,
+	CommitStatusWarning: 4,
```

Comment:
Not 100% sure about this, see previous code for reference:

https://github.com/go-gitea/gitea/pull/25839/files#diff-28c2d52083459d45d94ce384950fe4a3bddacdf125a296872a0b3f68734a1b16L27-R27

Reply:
Maybe 

CommitStatusWarning: 3,
CommitStatusSuccess: 4,

Reply:
Likely fixed with https://github.com/go-gitea/gitea/pull/27504/commits/89269758fe70a5f9d6986c8bed659c1d9fd8a246.

Reply:
The greater among statuses looks still wired.

Reply:
@delvh any opinion?

Reply:
I've simply compared it with the previous code, and I think a `warning` state is worse than a `pending` state as the `warning` state is guaranteed to be worse than a potentially successful `pending` state

Reply:
If I interpret the previous correctly, it should be ranked:

CommitStatusError
CommitStatusFailure
CommitStatusWarning
CommitStatusPending
CommitStatusSuccess

Reply:
Yes, that's exactly what I said

Reply:
I see the branch was already correct. Let's merge it.